diff --git a/auth-dialog/main.c b/auth-dialog/main.c
index a333aeb..9f49f95 100644
--- a/auth-dialog/main.c
+++ b/auth-dialog/main.c
@@ -1080,7 +1080,7 @@ static int validate_peer_cert(void *cbdata,
 #endif
 
 	certkey = g_strdup_printf ("certificate:%s:%d",
-				   openconnect_get_hostname(ui_data->vpninfo),
+				   openconnect_get_dnsname(ui_data->vpninfo),
 				   openconnect_get_port(ui_data->vpninfo));
 
 	accepted_hash = g_hash_table_lookup (ui_data->secrets, certkey);
@@ -1540,7 +1540,7 @@ static char *oc_server_url(auth_ui_data *ui_data)
 	 * This does mean that SNI-based proxies on the server end are going
 	 * to fail to find the right target, for older OpenConnect. */
 	return g_strdup_printf ("%s:%d",
-				openconnect_get_hostname(ui_data->vpninfo),
+				openconnect_get_dnsname(ui_data->vpninfo),
 				openconnect_get_port(ui_data->vpninfo));
 #endif
 }
